### PR TITLE
[feature] Indexer Test status filter

### PR DESCRIPTION
**Pull request by alecsg77**
Partialy solution for #3292

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -585,25 +585,26 @@ To get all Jackett indexers including their capabilities you can use `t=indexers
 
 ### Filter indexers
 
-Another special "filter" indexer is avaible at <code>/api/v2.0/indexers/<i><b>filter</b></i>/results/torznab</code>
-It will query the configured indexers that match the <i><b>filter</b></i> expression criterias and return the combined results as "all".
+Another special "filter" indexer is avaible at `/api/v2.0/indexers/<filter>/results/torznab`
```
Comment (ilike2burnthing) [inline README.md:588]:
```suggestion
Another special "filter" indexer is available at `/api/v2.0/indexers/<filter>/results/torznab`
```

**Comment by ilike2burnthing** [comment]
I'll test this out tomorrow once I've slept and have a clear head, but in the mean time if anyone else (@binhex) wants to have a go - [0.18.20](https://dev.azure.com/Jackett/0f0cb3d0-9b6b-4a86-876d-0c2e17c18801/_apis/build/builds/3454/artifacts?artifactName=drop&api-version=6.0&%24format=zip).

**Comment by binhex** [comment]
> but in the mean time if anyone else (@binhex) wants to have a go - [0.18.20](https://dev.azure.com/Jackett/0f0cb3d0-9b6b-4a86-876d-0c2e17c18801/_apis/build/builds/3454/artifacts?artifactName=drop&api-version=6.0&%24format=zip).

absolutely!, i will let you know how i get on.

**Comment by ilike2burnthing** [comment]
Works well. No comments here.

**Comment by alecsg77** [comment]
You can call directly Jackett from the browser using an URL like this
`http://<ip>:9117/api/v2.0/indexers/<filter>/results?apikey=<apiKey>&Query=<search>`
and see the results 😃 

`<filter>` will be `test:passed` or `test:failed` in this case.

**Comment by binhex** [comment]
@alecsg77 perfect!, i get it now, can you just confirm the below filter is the correct syntax, im just getting my head around combining multiple filters and not 100% sure on it:-

```tag:all,!type:private+lang:en+test:passed```

so does that equate to: all indexers, of type NOT private (aka public), language english only, and only indexers with a test of passed, correct?.

**Comment by alecsg77** [comment]
And operator have precedence over Or
so
indexers where (`Tags` contains `all`) OR (`Type` is not `private` and `Language` begin with `en` and `Test` is `passed`)

`Test is passed` -> `No Last Error`

**Comment by binhex** [comment]
ahh sorry i wasn't aware of the significance of ```,``` and ```+``` in the filter, thanks for that.

**Comment by binhex** [comment]
hmm so how can i AND all filters such that i can filter certain tagged index sites AND certain index site types AND language AND test passed?, e.g. filter (not working):-

```tag:all+type:public+lang:en+test:passed```

**Comment by ilike2burnthing** [comment]
Out of interest, is your `all` tag on all your configured indexers?

**Comment by ilike2burnthing** [comment]
I'm happy for this to be merged, however given that https://github.com/Jackett/Jackett/pull/11706 contains all the code for this PR, do you want to close this one and just focus on it?

**Comment by alecsg77** [comment]
No, I want to keep them separated and deliver this before the other one.

**Review by ilike2burnthing — APPROVED** [review]

**Comment by ilike2burnthing** [comment]
@binhex all working ok for you?

**Comment by binhex** [comment]
> @binhex all working ok for you?

it looks good from my side, it's actually already adding value for me as i had an outage on two of the indexers i have enabled and i could see they were marked as failed and the metadata downloader didn't seem to miss a beat :-), ignoring the 'failed' indexers, so im really happy with the change!.

**Comment by ilike2burnthing** [comment]
Thanks. This will hopefully be merged soon when garfield can take a look at it, so the next PR will be https://github.com/Jackett/Jackett/pull/11706 if you want to try that one.

**Review by garfield69 — APPROVED** [review]
LGTM
played around with windows Sonarr and Jackett with a variety of filters and got the expected results.

**Comment by garfield69** [comment]
v0.18.74

**Pull request merged**